src/main/kotlin/app/revanced/patches/duolingo/unlocksuper/patch/UnlockDuolingoSuperPatch.kt

```diff
@@ -0,0 +1,28 @@
+package app.revanced.patches.duolingo.unlocksuper.patch
+
+import app.revanced.extensions.toErrorResult
+import app.revanced.patcher.annotation.Compatibility
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Package
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.duolingo.unlocksuper.fingerprints.*
+
+@Patch
+@Name("Unlock Duolingo Super")
+@Description("Unlocks Duolingo Super features.")
+@Compatibility([Package("com.duolingo", versions = ["5.117.3"])])
```

Comment:
I don't think the version constraint is necessary here. The patch looks durable enough to withstand updates

Reply:
I missed that one, this hardcoded offset should be replaced with an index that is relative to an opcode pattern from the fingerprint.